src/cgr_gwas_qc/workflow/scripts/concordance_table.py

```diff
@@ -58,16 +69,65 @@ def read(filename: PathLike) -> pd.DataFrame:
     return pd.read_csv(filename, dtype=DTYPES)
 
 
-def build(filename: Path, concordance_threshold: float, pi_hat_threshold: float) -> pd.DataFrame:
+def _sort_ids(x: pd.DataFrame):
+    """ "Sort IDs alphanumerically."""
+    x.IID1, x.IID2 = np.where(x.IID1 < x.IID2, [x.IID1, x.IID2], [x.IID2, x.IID1])
+    x.rename(columns={"IID1": "ID1", "IID2": "ID2"}, inplace=True)
+    return x
+
+
+def prep_concordance_table(pi_hat_threshold: float, concordance_threshold: float, x: pd.DataFrame):
+    """ "Prepares the concordance_table by appending new columns based on thresholds."""
     return (
-        plink.read_genome(filename)
+        _sort_ids(x)
         .assign(is_ge_pi_hat=lambda x: x.PI_HAT >= pi_hat_threshold)
         .assign(concordance=lambda x: x.IBS2 / (x.IBS0 + x.IBS1 + x.IBS2))
         .assign(is_ge_concordance=lambda x: x.concordance >= concordance_threshold)
         .reindex(DTYPES.keys(), axis=1)
     )
 
 
+def process_genome_chunk(pi_hat_threshold: float, concordance_threshold: float, chunk):
+    """ "Processes each chunk of genome file."""
+    temp_file = tempfile.NamedTemporaryFile(delete=False)
+    prep_concordance_table(pi_hat_threshold, concordance_threshold, chunk).to_csv(
+        temp_file.name, index=False, header=False
+    )
+    return temp_file.name
+
+
+def build(
+    filename: Path,
+    concordance_threshold: float,
+    pi_hat_threshold: float,
+    outfile: Path,
+    threads: int,
+):
+
+    plink_genome_file = plink.read_genome(
+        filename, required_cols=["IID1", "IID2", "PI_HAT", "IBS0", "IBS1", "IBS2"], chunksize=100000
```

Comment:
Why do we specify a different `chunksize` here compared to the default size?

Reply:
Revised in 7ff778a298adf3f732ff0984bdba3870ef135dcf. I revised the default in read_genome and it will now use the default.